loki/transform/fortran_c_transform.py

```diff
@@ -64,7 +65,10 @@ def transform_file(self, sourcefile, **kwargs):
             self.transform_subroutine(routine, **kwargs)
 
     def transform_module(self, module, **kwargs):
-        path = Path(kwargs.get('path'))
+        if self.path is None:
```

Comment:
The `transform_module` method should never be invoked because the `item_filter` is not changed from the default to include anything beyond `SubroutineItem`:

https://github.com/ecmwf-ifs/loki/blob/3af8578c629cdb50b1c0c9ef7446a71eafa378e9/loki/transform/transformation.py#L78-L83

I suspect it is, however, used via direct invocation on files or in the test base?